robots/cmd/flake-report-creator/cmd/jenkins.go

```diff
@@ -411,8 +412,32 @@ func fetchCompletedBuildsForJob(startOfReport time.Time, lastBuildNumber int64,
 	return completedBuilds
 }
 
-// httpStatusOrDie fetches stringly typed error code produced by jenkins client or logs a fatal error if conversion to
-// int is not possible
+func getBuildWithRetry(job *gojenkins.Job, ctx context.Context, buildNumber int64, fLog *log.Entry) (build *gojenkins.Build, statusCode int, err error) {
+	retry.Do(
```

Comment:
By default, this will be retried 10 times.  Maybe 3 times is enough?

Reply:
I think I don't have enough data about this to agree to this. From what I saw this might be right. But if three retries are not enough then the whole report will be cancelled. So I'd say rather be safe than be sorry?

Reply:
BTW from the runs over the weekend I didn't see a single report timeone (which was what previously happened)

Reply:
Good to hear. I would say something is wrong with Jenkins if it takes more than 3 times. Nevertheless, let's merge this and adjust based on observation(maybe make it configurable).